src/Keystore/AESParameters.cpp

```diff
@@ -14,18 +14,30 @@ using namespace TW;
 
 namespace TW::Keystore {
 
-AESParameters::AESParameters() {
-    iv = Data(blockSize, 0);
-    random_buffer(iv.data(), blockSize);
+AESParameters::AESParameters(AESSize keyLength) : mKeyLength(keyLength) {
+    // https://www.reddit.com/r/crypto/comments/30x5xg/what_length_should_the_iv_be_for_aes256ctr/
+    // First off, AES 128 uses a 128 bit key.
+    // So if you're using AES 256, you're using a 256 bit key.
+    // Let's not confuse the block length with key length here.
+    // For AES, your block length is always going to be 128 bits/16 bytes regardless of the key length used
+    iv = Data(A128, 16);
+    random_buffer(iv.data(), 16);
 }
 
 namespace CodingKeys {
 static const auto iv = "iv";
 } // namespace CodingKeys
 
 /// Initializes `AESParameters` with a JSON object.
-AESParameters::AESParameters(const nlohmann::json& json) {
+AESParameters::AESParameters(const nlohmann::json& json, const std::string& cipher) {
     iv = parse_hex(json[CodingKeys::iv].get<std::string>());
+    if (cipher == "aes-128-ctr" || cipher == "aes-128-cbc") {
+        mKeyLength = A128;
+    } else if (cipher == "aes-192-ctr") {
+        mKeyLength = A192;
+    } else if (cipher == "aes-256-ctr") {
+        mKeyLength = A256;
+    }
```

Comment:
maybe add unknown cipher handling here?

Reply:
There is no possibility to have an unknown cipher, because it's from a constant defined enum - the cipher come from the generated json from this same enum.